integrations/vite/other-transforms.test.ts

```diff
@@ -0,0 +1,153 @@
+                  return code.replace(/red/g, 'blue')
+                }
+              },
+            },
+          ],
+        })
+      `,
+      'index.html': html`
+        <head>
+          <link rel="stylesheet" href="./src/index.css" />
+        </head>
+        <body>
+          <div class="foo">Hello, world!</div>
+        </body>
+      `,
+      'src/index.css': css`
+        @import 'tailwindcss/theme' theme(reference);
+        @import 'tailwindcss/utilities';
+
+        .foo {
+          color: red;
+        }
+      `,
+    },
+  }
+}
+
+for (let transformer of ['postcss', 'lightningcss'] as const) {
+  describe(transformer, () => {
+    test(`production build`, createSetup(transformer), async ({ fs, exec }) => {
+      await exec('pnpm vite build')
+
+      let files = await fs.glob('dist/**/*.css')
+      expect(files).toHaveLength(1)
+      let [filename] = files[0]
+
+      await fs.expectFileToContain(filename, [
+        css`
+          .foo {
+            color: blue;
+          }
+        `,
+      ])
+    })
+
+    test(`dev mode`, createSetup(transformer), async ({ spawn, getFreePort, fs }) => {
+      let port = await getFreePort()
+      await spawn(`pnpm vite dev --port ${port}`)
+
+      await retryAssertion(async () => {
+        let styles = await fetchStyles(port, '/index.html')
+        expect(styles).toContain(css`
+          .foo {
+            color: blue;
+          }
+        `)
+      })
+
+      await retryAssertion(async () => {
+        await fs.write(
+          'src/index.css',
+          css`
+            @import 'tailwindcss/theme' theme(reference);
+            @import 'tailwindcss/utilities';
+
+            .foo {
+              background-color: red;
+            }
+          `,
+        )
+
+        let styles = await fetchStyles(port)
+        expect(styles).toContain(css`
+          .foo {
+            background-color: blue;
+          }
+        `)
+      })
+    })
+
+    test('watch mode', createSetup(transformer), async ({ spawn, fs }) => {
+      await spawn(`pnpm vite build --watch`)
+
+      await retryAssertion(async () => {
+        let files = await fs.glob('dist/**/*.css')
+        expect(files).toHaveLength(1)
+        let [, styles] = files[0]
+
+        expect(styles).toContain(css`
+          .foo {
+            color: blue;
+          }
+        `)
+      })
+
+      await retryAssertion(async () => {
+        await fs.write(
+          'src/index.css',
+          css`
+            @import 'tailwindcss/theme' theme(reference);
+            @import 'tailwindcss/utilities';
+
+            .foo {
+              background-color: red;
+            }
+          `,
+        )
+
+        let files = await fs.glob('dist/**/*.css')
+        expect(files).toHaveLength(1)
+        let [, styles] = files[0]
+
+        expect(styles).toContain(css`
+          .foo {
+            background-color: blue;
+          }
+        `)
+      })
+    })
+  })
+}
```

Comment:
Have we decided on whether or not these transforms are expected to run on generated utilities? Can we add a test for that explicitly?

Reply:
It currently will, I don't know if we want to make this explicit by adding a test because this feels like something we might want to change later (with the two-pass approach).

I can add a test to document it tho and add a comment.

Reply:
yeah I'd add a test and a comment noting that we're okay with changing the behavior.